### PR TITLE
Use consistent inline notice style for deposit settings notices

### LINKED ISSUE #6424: Update `Settings > Deposits` notices to current inline notice style (e.g. `Your first deposit will be held for 7 days. Deposit scheduling will be available after this period.`
### Describe the bug
In `WooCommerce > Settings > Deposits`, the new account waiting period notice uses an old variation of the component. It should use ~`inline notice`~ `BannerNotice` component, notice no bar on left, and explicit `Learn more` instead of help icon. The goal of this issue is to update all similar notices in `Deposits` section of the settings.

Thanks to @elizaan36 for spotting this. 

Ideally we can just switch in a new component from Woo admin / gutenberg, maybe we're using custom styles or an old version of the component?

#### Should look like this
image-31

##### Currently looks like this
Screenshot 2023-06-02 at 10 58 13 AM

275905817-9e207a78-be66-4a0a-81e5-64c870d28a8e

### To Reproduce

1. `WooCommerce > Settings > Deposits`
2. Do whatever you need to do to show the new account notice – e.g. in test mode, or onboard a new WCPay account.

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #6424

#### Changes proposed in this Pull Request

This PR updates the style of the notices within the deposits sections of the setting screen to the preferred "Inline Notice", as per issue description.

Deposits blocked:

image

Account hasn't completed waiting period:

image

#### Testing instructions

* Use an account with deposits blocked and ensure the Payments → Settings "Deposit scheduling unavailable" notice renders as per screenshot above.
* Use an account with deposits enabled yet within the new account waiting period and ensure the Payments → Settings "First deposit will be held" notice renders as per screenshot above.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times.
- [x] Covered with tests **Yes**
- [x] Tested on mobile **Yes**

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8042` or branch name `fix/6424-update-settings-deposit-notice-ui` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/6424-update-settings-deposit-notice-ui) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 29c1ba0730be9cfc52b47d7d8189738c4bc21ee5
- Build time: 2024-01-19 00:07:51 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/deposits/index.js [hunk 1]
```diff
@@ -160,56 +156,47 @@ const DepositsSchedule = () => {
...
-						'woocommerce-payments'
-					) }
-				</span>
-				<a
-					aria-label={ __(
-						'Learn more about deposit scheduling.',
+			<InlineNotice status="warning" isDismissible={ false } icon>
+				{ interpolateComponents( {
+					mixedString: __(
+						'Deposit scheduling is currently unavailable for your store. {{learnMoreLink}}Learn more{{/learnMoreLink}}',
```
Comment (brucealdridge) [inline client/settings/deposits/index.js:162]: I'm guessing changing the link from `Learn more about deposit scheduling` to `Learn more` was about making it more consistent.

**Review by brucealdridge — APPROVED** [review]
Tested and reviewed code. Changes are more consistent. 👍

File: client/settings/deposits/index.js (hunk 1 above)
Comment (Jinksi) [reply client/settings/deposits/index.js:162]: Yeh, `Learn more about deposit scheduling` was the aria-label of the `?` icon button (not rendered as text) and seemed too verbose, given we usually use `Learn more` in these notices and the design in issue #6424 showed `Learn more` as well.

**Pull request merged**